modules/post/linux/gather/busybox_pingnet.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Post
+
+  include Msf::Post::File
+
+  def initialize
+    super(
+      'Name'         => 'BusyBox Ping Network',
+      'Description'  => 'This module will be applied on a session connected
+                         to a BusyBox sh shell. The script will ping a range of
+                         ip adresses from the router or device executing BusyBox.',
+      'Author'       => 'Javier Vicente Vallejo',
+      'License'      => MSF_LICENSE,
+      'References'   =>
+        [
+          [ 'URL', 'http://vallejo.cc']
+        ],
+      'Platform'      => ['linux'],
+       'SessionTypes'  => ['shell']
+    )
+
+    register_options(
+      [
+        OptAddress.new('IPRANGESTART',   [ true, "The first ip address of the range to ping.", nil ]),
+        OptAddress.new('IPRANGEEND',   [ true, "The last ip address of the range to ping.", nil ])
+      ], self.class)
+
+  end
+
+  def run
+
+    #this module will send a sh script for busybox shell for doing ping to a range of ip address from
+    #the router or device that is executing busybox. It could be possible to calculate each ip address
+    #of the range of ip addresses in the ruby script and execute each ping command with cmd_exec, but
+    #it would generate an unnecesary traffic in the connection with the busybox device (usually telnet)
```

Comment:
@jvazquez-r7 you are right. I thought about this when coding the module and i was not sure about the best option. If the script is not send, it will be necessary to send a command "ping -c 1 <ip>" for each address to ping. If there are more than 70 addresses to scan, it will generate more traffic than sending the script. Else, sending the script will generate more traffic. 

Finally I decided to send the script because the scan is faster with it instead of sending individual commands, and in addition, from my experience with some routers, i found some routers go crazy when they get too many commands continuously.